daemon/mgr/container_copy.go

```diff
@@ -0,0 +1,181 @@
+package mgr
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/chrootarchive"
+)
+
+// StatPath stats the dir info at the specified path in the container.
+func (mgr *ContainerManager) StatPath(ctx context.Context, name, path string) (stat *ContainerPathStat, err error) {
+	c, err := mgr.container(name)
+	if err != nil {
+		return nil, err
+	}
+	c.Lock()
+	defer c.Unlock()
+
+	err = mgr.Mount(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+	defer mgr.Unmount(ctx, c)
```

Comment:
Hi Yue,
Would you please take a look at [https://github.com/moby/moby/pull/39252](https://github.com/moby/moby/pull/39252) at your convenience? 
> cp command are vulnerable to a symlink-exchange attack with Directory Traversal, giving attackers arbitrary read-write access to the host filesystem with root privileges.

Reply:
I think the latter one we don't have risk, since we didn't support follow link now. But the former one issue in moby we have same risk. invite @allencloud @fuweid to see.

Reply:
I'm not sure it's necessary to do filesystem operations, I was wondering if it's possible to implement `cp` by using tar/untar like `kubectl cp`?

Reply:
@CodeJuan What kind of file operations you mean? seems there's a patch fix on chrootarchive package [#39292](https://github.com/moby/moby/pull/39292)

Reply:
`kubectl cp` doesn't work well when the container is in distroless mode, it also may frustrate by some malicious image with bad `tar binary` @CodeJuan 
 Actually, the solution of this issue